pkg/atlantis/README.md

```diff
@@ -0,0 +1,146 @@
+
+```yaml
+repos:
+  - id: /.*/
+    pre_workflow_hooks:
+      - run: ./repo-config-generator.sh
+```
+
+In `repo-config-generator.sh`, you need to first generate the varfiles for all components in all stacks,
+then generate `atlantis.yaml` repo config file using the following `atmos` commands:
+
+```shell
+atmos terraform generate varfiles --file-template=varfiles/{tenant}-{environment}-{stage}-{component}.tfvars.json
+
+atmos atlantis generate repo-config --config-template config-template-1 --project-template project-template-1 --workflow-template workflow-template-1
+```
+
+You can also run these commands manually and commit the generated varfiles and `atlantis.yaml` repo config.
+
+Note that the `-file-template` parameter in the `atmos terraform generate varfiles` command must correspond to the two settings in `atmos.yaml`:
+
+- `when_modified` must use the same template with the context tokens - this will allow Atlantis to check if any of the generated variables were
+  modified
+- workflow `extra_args` must use the same template with the context tokens - this will allow Atlantis to run Terraform commands with the
+  correct `-var-file` parameters
+
+Supported context tokens: `{namespace}`, `{tenant}`, `{environment}`, `{region}`, `{stage}`, `{component}`, `{component-path}`.
+
+```yaml
+# Integrations
+integrations:
+
+  # Atlantis
+  # https://www.runatlantis.io/docs/repo-level-atlantis-yaml.html
+  atlantis:
+    project_templates:
+      project-template-1:
+        # generate a project entry for each component in every stack
+        name: "{tenant}-{environment}-{stage}-{component}"
+        workspace: "{workspace}"
+        dir: "{component-path}"
+        terraform_version: v1.2
+        delete_source_branch_on_merge: true
+        autoplan:
+          enabled: true
+          when_modified:
+            - "**/*.tf"
+            - "varfiles/{tenant}-{environment}-{stage}-{component}.tfvars.json"
+
+    workflow_templates:
+      # generate a workflow entry for each component in every stack
+      workflow-template-1:
+        plan:
+          steps:
+            - init
+            - plan:
+                extra_args:
+                  - "-var-file"
+                  - "varfiles/{tenant}-{environment}-{stage}-{component}.tfvars.json"
+        apply:
+          steps:
+            - apply:
+                extra_args:
+                  - "-var-file"
+                  - "varfiles/{tenant}-{environment}-{stage}-{component}.tfvars.json"
+```
+
+Using the config, project and workflow templates, `atmos` generates a separate workflow for each `atlantis` project (which corresponds to an `atmos`
+component in a stack):
+
+```yaml
+version: 3
+automerge: true
+delete_source_branch_on_merge: true
+parallel_plan: true
+parallel_apply: true
+projects:
+  - name: tenant1-ue2-staging-test-test-component-override-3
+    workspace: test-component-override-3-workspace
```

Comment:
Since the workspaces are unique, couldn't we set the tfvar files to be the same as the workspace name ? That way, we could use the `$WORKSPACE` key in a single workflow without having to create a unique workflow per project.

Reply:
Hmmm so is it supported ? Could we use a single workflow across all the atlantis projects? If so, it would be good to add that as an example here and if people wanted a custom workflow per project, then that could be a second option. I think more would prefer the former since that would be simpler to manage and reduce the configuration size.